src/transformers/cache_utils.py

```diff
@@ -813,7 +813,8 @@ def _quantize(self, tensor, axis):
         if is_optimum_quanto_available():
             from optimum.quanto import quantize_weight
 
-            qtensor = quantize_weight(tensor, self.qtype, axis, self.q_group_size)
```

Comment:
Can you have a look @zucchini-nlp as you did the change in this [PR](https://github.com/huggingface/transformers/pull/34000/files). Looking at optimum-quanto source code, `quantize_weight` do require to pass `scale`.

Reply:
Now when I look I think it is related to the version of optimum quanto. I see pre v0.24 had no scale

https://github.com/huggingface/optimum-quanto/blob/832f7f5c3926c91fe4f923aaaf037a780ac3e6c3/optimum/quanto/tensor/qweight.py#L32-L39

But after v0.24 we have to pass scale in before group size 

https://github.com/huggingface/optimum-quanto/blob/f3c400e9b5b28b499f87c30325f8628d50417eef/optimum/quanto/tensor/weights/quantization.py#L27-L36

@SunMarc if that is correct, prob we need to check the version also. Seems like a lot of checks but since the old `quanto` should be removed in next v4.47 release, could be a workaround. WDYT?

Reply:
You mean the prev PR I merged? It only made the code compatible with optimum-quanto v0.24 but I forgot there could be older versions. For the why optimum-quanto is changing its code, i have no idea. But would be nice if they wouldnt change it drastically anymore 😅 

i guess you'll have more info about future maintaining plans in quanto :)

Reply:
@w3rew thanks for opening the PR, can you please update with suggested changes?

Reply:
Sure! Will look into it shortly.

Reply:
cc @dacorvo for visibility

Reply:
@zucchini-nlp `optimum-quanto` 0.2.5 release was synchronized with the switch from `quanto` to `optimum-quanto` in `transformers` at the beginning of october, and the code in `cache_utils.py` was correct.
It is your pull-request to align with 0.2.4 (a version that was never supported by `transformers` that was actually incorrect.

Reply:
@zucchini-nlp  So, just to clarify: is the version check necessary? If the code used to support only 0.2.5, the additional check seems to be redundant. More so if 0.2.4 was never supported.

Reply:
@w3rew hmm, iirc the first time I triggered tests for quant cache `pip` installed me the `0.2.4` version by default so that was the reason for modification. In case as you mention, we can leave only the post-0.2.5 logic but would be nice to add a version check near the import-chech for optimum quanto with explicit error message that we need the optimum quanto with v0.2.5 or higher